### PR TITLE
Prevent double logging

### PR DESCRIPTION
**Pull request by maartenplieger**
Fixed two issues since we are using fast-api:
- Logging showed up twice
- Process had no timeout

### CODE REVIEW

**Comment by belentorrente** [comment]
Looks good to me.

File: python/lib/adaguc/CGIRunner.py [hunk 1]
```diff
@@ -37,8 +37,14 @@ def run(self, cmds, url, output, env=[],  path=None, isCGI=True):
         # Execute adaguc-server binary
         ON_POSIX = 'posix' in sys.builtin_module_names
         process = Popen(cmds, stdout=PIPE, stderr=PIPE, env=localenv, close_fds=ON_POSIX)
-        (processOutput, processError) = process.communicate()
- 
+        
+        try:
+            (processOutput, processError) = process.communicate(timeout=timeout)
+        except TimeoutExpired:
+            process.kill()
```
Comment (lukas-phaf) [inline python/lib/adaguc/CGIRunner.py:44]: According to [this](https://docs.python.org/3/library/subprocess.html#subprocess.Popen.communicate), you also need to do a `proc.communicate()` to do a proper cleanup?

File: python/python_fastapi_server/configure_logging.py [hunk 2]
```diff
@@ -3,7 +3,17 @@
 """Configures logging for the adaguc-server python wrapper"""
 import sys
 
+def run_once(f):
+    def wrapper(*args, **kwargs):
+        if not wrapper.has_run:
+            wrapper.has_run = True
```
Comment (lukas-phaf) [inline python/python_fastapi_server/configure_logging.py]: Change the order like [here](https://gist.github.com/jasongrout/3804691).
If `configure_logging` somehow fails, it will not run again, as `has_run` is set to True before it is run.

File: python/python_fastapi_server/routers/wmswcs.py [hunk 3]
```diff
@@ -46,6 +46,7 @@ async def handle_wms(req: Request, ):
     # Run adaguc-server
     status, data, headers = adaguc_instance.runADAGUCServer(query_string,
                                                             env=adagucenv,
+                                                            showLogOnError=False,
```
Comment (lukas-phaf) [inline python/python_fastapi_server/routers/wmswcs.py:49]: Was this change intended?

File: python/python_fastapi_server/configure_logging.py [hunk 4]
```diff
@@ -3,7 +3,17 @@
...
+            return f(*args, **kwargs)
 
+    wrapper.has_run = False
+    return wrapper
+
+
+@run_once
 def configure_logging(logging):
```
Comment (lukas-phaf) [inline python/python_fastapi_server/configure_logging.py:6]: Not clear to me why it is called twice. The solution is a bit... brutal.

File: python/python_fastapi_server/routers/wmswcs.py (hunk 3 above)
Comment (maartenplieger) [reply python/python_fastapi_server/routers/wmswcs.py:49]: Yes, the logging is in wmswcs.py always displayed, see line 57. It does not need to show it once more when there is an error.

File: python/lib/adaguc/CGIRunner.py (hunk 1 above)
Comment (maartenplieger) [reply python/lib/adaguc/CGIRunner.py:44]: yes, good one. Clean kill.

File: python/python_fastapi_server/configure_logging.py (hunk 2 above)
Comment (maartenplieger) [reply python/python_fastapi_server/configure_logging.py]: Okay, I think I added the @wraps correctly.

File: python/python_fastapi_server/configure_logging.py (hunk 4 above)
Comment (maartenplieger) [reply python/python_fastapi_server/configure_logging.py:6]: I found that gunicorn or uvicorn also add a logging handler.

I tried to add logging.getLogger().handlers.clear() in configure_logging.

This looks like a better solution to me?

File: python/python_fastapi_server/configure_logging.py (hunk 2 above)
Comment (maartenplieger) [reply python/python_fastapi_server/configure_logging.py]: Now completely removed the run_once part.

**Review by lukas-phaf — APPROVED** [review]
